PATTERN.md

```diff
@@ -14,34 +14,79 @@ Request-derived Context.
 and serverless functions.
-As long as it's receiving HTTP requests and needs context, the pattern is applicable.
+As long as it is receiving HTTP requests and needs context, the pattern is applicable.
 
 This pattern is similar to the **Request Context** concern found in many frameworks, with the key difference being
 that the context is derived from the request itself.
 All request-derived context is request context, but not all request context is request-derived context.
 
 ## Problem
 
+Modern web applications often rely on contextual information, and because of how the HTTP request/response model works,
+there is no guarantee that the context will be relevant for more than the current request.
+The context could be anything, such as the identity of the user making the request, the tenant on whose behalf the 
+request is made, or the region or language the request is being made in.
+It could also possibly be required across multiple requests, with each request possibly manipulating the context.
+This is a problem because it requires state, and HTTP is a stateless protocol.
+
+We have long since developed methods of adding state on top of HTTP, such as cookies, or more appropriately, sessions.
+However, for a web application to know the state that is relevant to the current request, it needs context, bringing us
+full circle.
+While an HTTP request does not carry state or contain context, it does carry information that can be used to derive it.
+As well as allowing for arbitrary data to be passed along with the request (e.g. custom headers, query parameters, or 
+cookies), which can be used in the same way.
+
+For the context to be useful, the relevant data needs to be extracted from the request, resolved to whatever its final
+form is, and made available for the duration of the request.
+Without any form of defined structure, the logic that handles this is typically scattered throughout the codebase, with
+the same concept being implemented in multiple places, leading to code duplication and a lack of consistency.
+
+While this behaviour is common across modern web frameworks, it is rarely identified or treated as a formal
+architectural concern.
+This means we lack terminology to describe it, making it difficult to draw parallels between different components
+within an application that ostensibly requires the same functionality, whether entirely or in part.
+
 ## Force
 
+There are several forces at play that make this pattern necessary:
+
+- **Statelessness** <br/>
+  Each HTTP request is handled in isolation; therefore, it is without shared memory or persistent context.
+  This means the context must be derived from the request itself.
+- **Early Context Requirements** <br/>
+  Core decisions at both the application and domain level will often depend on context, which means it must be
+  resolved early on, so the request can be processed correctly.
```

Comment:
"resolved" is a funny word, because it could be understood to mean completely understood; whereas multi-request context _requires_ that it isn't finalised. Think "Cookes are resolved" which means they're fully decoded, but does not mean they are appended to the response.

Not sure what the better wording is, though. Context is recognised? Loaded?